src/Symfony/Bundle/SecurityBundle/Form/Type/UsernamePasswordType.php

```diff
@@ -42,31 +41,23 @@ public function __construct(RequestStack $requestStack)
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
         $builder
-            ->add('username', 'Symfony\Component\Form\Extension\Core\Type\TextType')
-            ->add('password', 'Symfony\Component\Form\Extension\Core\Type\PasswordType')
+            ->add('_username', 'Symfony\Component\Form\Extension\Core\Type\TextType')
```

Comment:
> However I wonder if we can take advantage of the new FirewallConfig class to get the right field names dynamically instead of hardcoding _username, _password and _csrf_token values in the type, that would be cool!

That's probably too much for the need IMHO.
This will probably require defining parameters in the container from the `Symfony\Bundle\SecurityBundle\DependencyInjection\Security\Factory\FormLoginFactory`, inject the `FirewallMap` inside the `UserLoginType` along with the `RequestStack`, retrieve the firewall name and guess the right parameter to use.